src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -414,33 +432,21 @@ class GraphFrame private(
 
   // ========= Motif finding (private) =========
 
-
   /**
    * Primary method implementing motif finding.
-   * This recursive method handles one pattern (via [[findIncremental()]] on each iteration,
+   * This iterative method handles one pattern (via [[findIncremental()]] on each iteration,
    * augmenting the `DataFrame` in prevDF with each new pattern.
    *
-   * @param prevPatterns  Patterns already handled
-   * @param prevDF  Current DataFrame based on prevPatterns
-   * @param prevNames Current sequence of column names in the order as specified by prevPatterns
-   *                  For instance, `"(a)-[e]->(b)"` is Seq("a", "e", "b")
-   *                  `"(a)-[e]->(b); (b)-[]->(c)"` is Seq("a", "e", "b", "c")
-   * @param remainingPatterns  Patterns not yet handled
-   * @return `DataFrame` augmented with the next pattern, or the previous DataFrame if done
-   *        Seq[String] sequence of column names for the `DataFrame` appended to in order
-   *        as specified by the next pattern
-   */
-  private def findSimple(
-      prevPatterns: Seq[Pattern],
-      prevDF: Option[DataFrame],
-      prevNames: Seq[String],
-      remainingPatterns: Seq[Pattern]): (DataFrame, Seq[String]) = {
-    remainingPatterns match {
-      case Nil => (prevDF.getOrElse(sqlContext.emptyDataFrame), prevNames)
-      case cur :: rest =>
-        val (df, names) = findIncremental(this, prevPatterns, prevDF, prevNames, cur)
-        findSimple(prevPatterns :+ cur, df, names, rest)
-    }
+   * @return `DataFrame` containing all instances of the motif specified by the given patterns
+   */
+  private def findSimple(patterns: Seq[Pattern]): DataFrame = {
```

Comment:
This is an extraneous change.  Let me know if you want me to move it to a new PR.

Reply:
I feel like it's more readable to avoid recursion and reduce the number of arguments in the (internal) API.  It's weird to have outside users pass in initialization values which they shouldn't need to know about.  (I originally made this change when I thought I needed to call findSimple multiple times from find and wanted to simplify it; otherwise I wouldn't have bothered.)